controllers/helpers.go

```diff
@@ -642,3 +644,77 @@ func hasPEClusterServiceEnabled(peCluster *nutanixClientV3.ClusterIntentResponse
 	}
 	return false
 }
+
+// GetGPUDeviceIdList returns a list of GPU device IDs for the given list of GPUs
+func GetGPUList(ctx context.Context, client *nutanixClientV3.Client, gpus []infrav1.NutanixGPU, peUUID string) ([]*nutanixClientV3.VMGpu, error) {
+	resultGPUs := make([]*nutanixClientV3.VMGpu, 0)
+	for _, gpu := range gpus {
+		foundGPU, err := GetGPU(
+			ctx,
+			client,
+			peUUID,
+			gpu,
+		)
```

Comment:
nit: this can be compressed into one line.

Reply:
Fixed